lnwallet/channel.go

```diff
@@ -2010,7 +2017,6 @@ func (lc *LightningChannel) closeObserver(channelCloseNtfn *chainntnfs.SpendEven
 	}
 
 	lc.Lock()
-	defer lc.Unlock()
```

Comment:
add comment on why defer is not used

Reply:
definitely a good idea, the locking in this section is pretty tricky

Reply:
Seems like it should be restored? This way we can skip all the unlocks below and possibly avoid a double unlock.

Reply:
we need to release the lock while sending on the channels to avoid deadlocks